Makefile

```diff
@@ -191,12 +196,20 @@ scorecard-tests: operator-sdk
 # buildx is used to ensure same results for arm based systems (m1/2 chips)
 .PHONY: container
 container:
+ifeq (true,$(SKIP_DEPLOYMENT))
+	echo "Skipping container operator build since SKIP_DEPLOYMENT is true"
+else
 	docker buildx build --load --platform linux/${ARCH} -t ${IMG} --build-arg VERSION_PKG=${VERSION_PKG} --build-arg VERSION=${VERSION} --build-arg VERSION_DATE=${VERSION_DATE} --build-arg OTELCOL_VERSION=${OTELCOL_VERSION} --build-arg TARGETALLOCATOR_VERSION=${TARGETALLOCATOR_VERSION} --build-arg AUTO_INSTRUMENTATION_JAVA_VERSION=${AUTO_INSTRUMENTATION_JAVA_VERSION}  --build-arg AUTO_INSTRUMENTATION_NODEJS_VERSION=${AUTO_INSTRUMENTATION_NODEJS_VERSION} --build-arg AUTO_INSTRUMENTATION_PYTHON_VERSION=${AUTO_INSTRUMENTATION_PYTHON_VERSION} --build-arg AUTO_INSTRUMENTATION_DOTNET_VERSION=${AUTO_INSTRUMENTATION_DOTNET_VERSION} .
+endif
 
 # Push the container image, used only for local dev purposes
 .PHONY: container-push
 container-push:
+ifeq (true,$(SKIP_DEPLOYMENT))
+	echo "Skipping container operator push since SKIP_DEPLOYMENT is true"
```

Comment:
Isnt `container-push` != `deployment` anyway?

Reply:
It is but: if you want to skip the deployment and the build was skipped, it doesn't make sense to push the container image to the repository, right?

Reply:
In that case the checksum would stay the same anyway and not much would happen.

Personally, when I read `SKIP_DEPLOYMENT=true make ... container-push` I expect a new build in my registry and that no rollout will be executed.

But I am more concerned that we add logic that undermines explicit parameters.

Like here:
```
SKIP_DEPLOYMENT=true make ... container-push
# is equal to
make ...
```

Reply:
But if I didn't build the image, I would not have any image to push. That is an example if I want to run the tests about a release, for instance, where I will use the image provided by the manifest.

if the problem is the name of the variable, could you provide a better name? Or: should I create 3 variables (one to skip deployment, another to skip build and another to skip push??

Reply:
Actually i still do not see the need for it. Why can't the desired make rule be removed as an argument?

Thats what i tried to say in the comment above. Instead of `SKIP_SOMETHING_1=true make something1 something2` we could call `make something2` right?

Reply:
Ok, now I see your point.
The problem is: if the `prepare-e2e` rule is modified, we need to modify the developer's workflow. While, with this, it is not needed.

Reply:
>  The problem is: if the prepare-e2e rule is modified, we need to modify the developer's workflow. While, with this, it is not needed.

But why?

---

Whats your thoughts on that @pavolloffay @rubenvp8510 ?

Reply:
It's ok. I'll close the PR since is not as important and we're spending too much time on it.